src/frontend/packages/semantic-data-provider/src/hooks/useSparqlQuery.ts

```diff
@@ -0,0 +1,50 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { useGetIdentity, fetchUtils } from 'react-admin';
+
+const useSparqlQuery = (sparqlQuery?: string) => {
+  const { identity } = useGetIdentity();
+  const [inQuery, setInQuery] = useState(false);
+  const [status, setStatus] = useState<number | undefined>(undefined);
+  const [result, setResult] = useState<object | undefined>(undefined);
+
+  const sparqlEndpoint = useMemo(() => {
+    if (identity?.webIdData) {
+      return String(identity.webIdData?.endpoints?.['void:sparqlEndpoint'] || `${identity.id}/sparql`);
+    }
+    return null;
+  }, [identity]);
+
+  const fetch = useCallback(
+    async (query: string) => {
+      if (!sparqlEndpoint) throw new Error('No sparql endpoint available.');
+
+      const token = localStorage.getItem('token');
+
+      setInQuery(true);
+      const jsonResponse = await fetchUtils.fetchJson(sparqlEndpoint, {
```

Comment:
If you get the dataProvider (via `useDataProvider`), you have access to a `fetch` method that adds the token to the header, as well as the `Accept` header.

Reply:
Oops, I didn't mean to check this in anyways but thanks!